build.gradle

```diff
@@ -38,11 +38,16 @@ gradlePlugin {
 
     automatedPublishing = true
 
+    def releaseNotes = layout.projectDirectory.file("release/changes.md").asFile.text
```

Comment:
Is this change build caching compatible, meaning changes to the release notes will retrigger any required tasks to rerun (if any)?

I'm not saying there are cacheable tasks depending on this input, but maybe there are, and I want to be sure it was considered. (it needs to be muscle-memory to see any build change in the context of build caching and configuration caching)

Reply:
Good point - it looks like all tasks that need to be re-executed when `changes.md` changes are re-executed. So I think we are good here.

Reply:
Since this is reading a file at configuration time, then this is not compatible with configuration caching.

Reply:
I recommend reaching out to the build tool team to see what they recommend as far as setting release notes from a file.

Reply:
Yes, the tasks themselves will always re-execute but changes to the `changes.md` file will not invalidate the configuration cache. Try this:

1. Run `./gradlew clean`
2. Run `./gradlew generatePomFileForCCUD`
3. Inspect the description in `build/publications/commonCustomUserDataPluginMarkerMaven/pom-default.xml`
4. Modify `release/changes.md`
5. Run `./gradlew generatePomFileForCCUD`
6. Inspect the description in `build/publications/commonCustomUserDataPluginMarkerMaven/pom-default.xml` and you will see it did not change.

Reply:
Ah I see - I had been testing this exact scenario but with `publishToMavenLocal`. When invoked with that task, the description does get updated.

Reply:
Very nice. 😎 

https://ge.solutions-team.gradle.com/s/wogvc5hp3e6fs/console-log?page=1#L1